### PR TITLE
Issue #594 Explicit error message when LocalPersistenceService is req…

### PR DESCRIPTION
**Pull request by ljacomet**
…uired but missing

### CODE REVIEW

**Comment by lorban** [comment]
I'd consider an integration test as well.

File: core/src/main/java/org/ehcache/EhcacheManager.java [hunk 1]
```diff
@@ -264,6 +264,10 @@ private boolean isDiskTransient(Ehcache<?, ?> ehcache) {
     if (config.getResourcePools().getResourceTypeSet().contains(ResourceType.Core.DISK)) {
       LocalPersistenceService persistenceService = serviceLocator.getService(LocalPersistenceService.class);
       
+      if (persistenceService == null) {
+        throw new IllegalStateException("No LocalPersistenceService could be found - did you configure it at the CacheManager level?");
+      }
+
```
Comment (AbfrmBlr) [inline core/src/main/java/org/ehcache/EhcacheManager.java]: Can't we validate in builder ?

Comment (lorban) [inline core/src/main/java/org/ehcache/EhcacheManager.java]: I don't think we want our builders to have any kind of intelligence w.r.t the logical validity of what they're building.

Comment (ljacomet) [inline core/src/main/java/org/ehcache/EhcacheManager.java]: We cannot and should not validate in builder.

**Pull request merged**